frontend/src/components/TimetableGrid.tsx

```diff
@@ -0,0 +1,238 @@
+        }
+      }
+      if (flag) {
+        nonNullColumns = j;
+      } else {
+        break;
+      }
+    }
+    setDisplayCols(nonNullColumns);
+
+    return grid;
+  }, [timetableDetailsSections, isVertical]);
+
+  return (
+    <div className="ml-4 flex w-full">
+      {isVertical ? (
+        <div />
+      ) : (
+        <div
+          className={`grid items-center pr-2 text-lg text-center font-bold text-foreground/90 ${
+            isVertical
+              ? `grid-cols-${displayRows}`
+              : `grid-rows-${displayRows} mt-12`
+          }`}
+        >
+          {daysOfWeek.map((e) => (
+            <span key={e}>{e}</span>
+          ))}
+        </div>
+      )}
+      <div className="flex flex-col w-full">
+        {isVertical ? (
+          <div
+            className={`grid items-center pr-2 text-lg text-center font-bold text-foreground/90 ${
+              isVertical
+                ? `grid-cols-${displayRows}`
+                : `grid-rows-${displayRows} mt-12`
+            }`}
+          >
+            {daysOfWeek.map((e) => (
+              <span key={e}>{e}</span>
+            ))}
+          </div>
+        ) : (
+          <div
+            className={`grid justify-between text-md text-center font-bold text-foreground/90 ${
+              isVertical
+                ? `grid-rows-${displayCols}`
+                : `grid-cols-${displayCols}`
+            } `}
+          >
+            {timings
+              .filter((e, i) => i < displayCols)
+              .map((e) => (
+                <div className="mb-4 flex flex-col" key={e}>
+                  <span>{e}</span>
+                </div>
+              ))}
+          </div>
+        )}
+        <div
+          className={`grid w-full ${
+            isVertical
+              ? `grid-cols-${displayRows} grid-rows-${displayCols}`
+              : `grid-cols-${displayCols} grid-rows-${displayRows}`
+          }`}
+        >
+          {timetableGrid
+            .filter((e, i) =>
+              isVertical ? i < 6 * displayCols : i % 11 < displayCols,
+            )
+            .map((e, i) =>
+              e !== null ? (
+                <Tooltip delayDuration={100} key={2 * i}>
+                  <TooltipTrigger asChild>
+                    {/* biome-ignore lint/a11y/useKeyWithClickEvents: <explanation> */}
+                    <div
+                      className={`bg-primary-foreground border cursor-pointer transition duration-200 ease-in-out text-foreground/65 p-1.5 ${
+                        isVertical ? "" : "min-h-20"
+                      }`}
+                      onClick={(event) => handleUnitClick(e, event)}
+                    >
+                      <div className="relative flex h-full text-xs sm:text-sm flex-col justify-end bg-muted-foreground/30 p-1.5 rounded gap-0.5">
+                        <X
+                          size={16}
+                          className="absolute top-1 right-1 sm:visible invisible hover:stroke-[#EF4444]"
```

Comment:
```suggestion
                          className="absolute top-1 right-1 sm:visible invisible hover:stroke-destructive dark:hover:stroke-red-400 transition duration-100 ease-in-out"
```
Avoid use of custom colors, and add a slight transition.